### PR TITLE
APP-4686 :- Added regex for company name

### PR DESCRIPTION
**Pull request by prabhat-inss**

### CODE REVIEW

**Review by WayneBusby — CHANGES_REQUESTED** [review]
Can we add some unit tests that check the '&' does not cause any issues.  Its well known that & is a delimiter for query string values, which means that it may require url encoding before processing.

File: INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@ namespace INSS.FIP.Web.ViewModels;
 public class SearchParametersViewModel : IValidatableObject
 {
     private const string RegExForName = "^[a-zA-Z ]+(([',.\\-][a-zA-Z ])?[a-zA-Z ]*)*$";
+    private const string RegExForCompanyName = "^[a-zA-Z ]+(([',&.\\-][a-zA-Z ])?[a-zA-Z ]*)*$";
```
Comment (VasInsolvencyService) [inline INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs]: The ticket says , allow 100 characters including & and £ , Cant see £ in the RegEx

File: INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs [hunk 2]
```diff
@@ -31,7 +32,7 @@ public class SearchParametersViewModel : IValidatableObject
     public string? LastName { get; set; }
 
     [StringLength(100, ErrorMessage = StringLengthValidationError)]
-    [RegularExpression(RegExForName, ErrorMessage = InvalidCharactersValidationError)]
+    [RegularExpression(RegExForCompanyName, ErrorMessage = InvalidCharactersValidationError)]
```
Comment (VasInsolvencyService) [inline INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs]: Could you add unit test for the fix

File: INSS.FIP.Web.UnitTests/ValidationTests/SearchParametersViewModelValidationTests.cs [hunk 3]
```diff
@@ -2,20 +2,30 @@
 using INSS.FIP.Services;
 using INSS.FIP.Web.ViewModels;
 using Xunit;
-
+using INSS.FIP.Web.ViewModels;
```
Comment (VasInsolvencyService) [inline INSS.FIP.Web.UnitTests/ValidationTests/SearchParametersViewModelValidationTests.cs]: remove unnecessary namespace  (INSS.FIP.Web.ViewModels added twice)

File: INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs (hunk 1 above)
Comment (prabhat-inss) [reply INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs]: Refer the updated ticket ' £' is not required. and yes characters length could be of max 100. but this is already in place so don't require any code changes.

File: INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs (hunk 2 above)
Comment (prabhat-inss) [reply INSS.FIP.Web/ViewModels/SearchParametersViewModel.cs]: Added some unit test cases for this and improved the existing one as well.

File: INSS.FIP.Web.UnitTests/ValidationTests/SearchParametersViewModelValidationTests.cs (hunk 3 above)
Comment (prabhat-inss) [reply INSS.FIP.Web.UnitTests/ValidationTests/SearchParametersViewModelValidationTests.cs]: Removed in my last commit .

**Review by VasInsolvencyService — APPROVED** [review]

**Review by WayneBusby — APPROVED** [review]

**Pull request merged**